src/services/puzzlebang/puzzlebang-router.ts

```diff
@@ -0,0 +1,40 @@
+import { Router } from "express";
+import { StatusCodes } from "http-status-codes";
+import { Database } from "../../database";
+import RoleChecker from "../../middleware/role-checker";
+import { Role } from "../auth/auth-models";
+import { PuzzlebangCompleteRequestValidator } from "./puzzlebang-validators";
+
+const puzzlebangRouter = Router();
+
+puzzlebangRouter.post(
+    "/",
+    RoleChecker([Role.Enum.PUZZLEBANG]),
+    async (req, res, next) => {
+        try {
+            const requestInfo = PuzzlebangCompleteRequestValidator.parse(
+                req.body
+            );
+
+            const attendeeData = await Database.ATTENDEE.findOneAndUpdate(
+                { email: requestInfo.email },
+                { $addToSet: { puzzlesCompleted: requestInfo.puzzleId } },
+                { new: false }
+            );
+
+            if (!attendeeData) {
+                return res.sendStatus(StatusCodes.NOT_FOUND);
+            }
+
+            if (attendeeData.puzzlesCompleted.includes(requestInfo.puzzleId)) {
```

Comment:
Wouldn't this always be true?  Since you're calling `findOneAndUpdate` with `$addToSet` and adding the puzzle ID there?

Also, I think we can leave this out completely, since it looks like `$addToSet` ([if I'm reading this right](https://www.mongodb.com/docs/manual/reference/operator/update/addToSet/)) does nothing to the array if the value already exists in the set.

Reply:
Chatted offline, can ignore this comment (I didn't understand that `findOneAndUpdate` returns the document before the update was applied).